### PR TITLE
Bsearch doc for Array and Range

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This PR creates doc/bsearch.rdoc to provide common documentation for bsearch in Array and Range.

### CODE REVIEW

**Review by jeremyevans — APPROVED** [review]
Looks good, thanks!

File: doc/bsearch.rdoc [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  r.bsearch {|i| a[i] >= 4 } #=> 1
+  r.bsearch {|i| a[i] >= 6 } #=> 2
+  r.bsearch {|i| a[i] >= 8 } #=> 3
+  r.bsearch {|i| a[i] >= 100 } #=> nil
+  r = (0.0...Float::INFINITY)
+  r.bsearch {|x| Math.log(x) >= 0 } #=> 1.0
+
+These blocks make sense in find-minimum mode:
+
+  a = [0, 4, 7, 10, 12]
```
Comment (JuPlutonic) [inline doc/bsearch.rdoc:65]: 65 line need to be put in 55 line

**Comment by BurdetteLamar** [comment]
Array a is defined in line 49.

    On Monday, September 20, 2021, 12:34:37 PM CDT, 0x639d1dec74 ***@***.***> wrote:  
 
 


@JuPlutonic commented on this pull request.

In doc/bsearch.rdoc:
> +  a.bsearch {|x| x >= 4 } # => 4
+  a.bsearch {|x| x >= 6 } # => 7
+  a.bsearch {|x| x >= -1 } # => 0
+  a.bsearch {|x| x >= 100 } # => nil
+
+  r = (0...a.size)
+  r.bsearch {|i| a[i] >= 4 } #=> 1
+  r.bsearch {|i| a[i] >= 6 } #=> 2
+  r.bsearch {|i| a[i] >= 8 } #=> 3
+  r.bsearch {|i| a[i] >= 100 } #=> nil
+  r = (0.0...Float::INFINITY)
+  r.bsearch {|x| Math.log(x) >= 0 } #=> 1.0
+
+These blocks make sense in find-minimum mode:
+
+  a = [0, 4, 7, 10, 12]

65 line need to be put in 55 line

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by JuPlutonic** [comment]
Thanks!

**Pull request merged**